backend/api/serializers/MeetEventSerializer.py

```diff
@@ -5,11 +5,12 @@ class MeetEventSerializer(serializers.ModelSerializer):
     name = serializers.CharField(max_length=50, read_only=True)
     group_name = serializers.ReadOnlyField(source="group.name")
     event_type_name = serializers.ReadOnlyField(source="event_type.name")
+    num_heats = serializers.IntegerField(source="heat_event.count", read_only=True)
```

Comment:
Currently, heat_event.count returns the total number of records in the heat table, but since multiple records can have the same heat number (due to different lanes), this doesn't accurately reflect the number of heats. To get the correct number of heats, we need to calculate the maximum value of num_heat for the given event.

Reply:
Got it, I got confused with this value. 

As discussed, the new approach is to include the total number of heats in the Meet Event model since this value is being used in different parts of the code. This way, the value will be updated once the heats are created or deleted and directly accessed instead of being computed every time we retrieve the list of event, heats or heats by lane.